graphics/ImageMagick7/Portfile

```diff
@@ -3,6 +3,7 @@
 PortSystem          1.0
 PortGroup           conflicts_build 1.0
 PortGroup           github 1.0
+PortGroup           legacysupport 1.1
```

Comment:
While it is not required, but if we know what exactly fails without `legacysupport`, perhaps worth setting `legacysupport.newest_darwin_requires_legacy`.

Reply:
> if we know what exactly fails ...

Okay.  I threw in PG `legacysupport` on a total guess, to try to fix 10.6 builds.  Exactly what fails is incomplete `TargetConditionals.h`, from the builder logs.  It is the same for both x386 and x86_64.  Specifically, all three of these symbols are missing:
```
MagickCore/delegate.c:86:7: error: 'TARGET_OS_IOS' is not defined, evaluates to 0 [-Werror,-Wundef-prefix=TARGET_OS_]
  #if TARGET_OS_IOS || TARGET_OS_WATCH || TARGET_OS_TV
  etc.
```
I do not know whether plain `legacysupport` will fix this.  I would be delighted if you can suggest a better approach, or a more targeted fix.

Reply:
Max Version Needing Feature OSX10.10
https://github.com/macports/macports-legacy-support

Reply:
Yes, there it is, `TargetConditionals.h` with the desired missing definitions.  So, okay to include that `legacysupport` portgroup, the way I have it now?  If there is a way to target that specific header, please show me how.

Reply:
@Dave-Allured I just meant we could add `legacysupport.newest_darwin_requires_legacy 14`

Reply:
@barracuda156 We posted at almost exactly the same time.  Yes, based on the last builder results, this particular legacy fix is needed only up to 10.6.  Let's see how this one goes.